test/e2e_node_kind_source_map_test.ts

```diff
@@ -0,0 +1,625 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+// tslint:disable:no-unused-expression mocha .to.be.empty getters.
+
+import {expect} from 'chai';
+import * as path from 'path';
+import * as ts from 'typescript';
+
+import * as tsickle from '../src/tsickle';
+
+import {compile, getLineAndColumn} from './test_support';
+
+describe('source maps each node with transformer', () => {
+  it('maps at the start of lines correctly', () => {
+    const sources = new Map([[
+      'input.ts', `let x : number = 2;
+      x + 1;
+      let y = {z: 2};
+      y.z;`
+    ]]);
+
+    const {compiledJS, sourceMap} = compile(sources, {useTransformer: true});
+
+    {
+      const {line, column} = getLineAndColumn(compiledJS, 'var /** @type {number} */ x');
```

Comment:
These test are overall a bit boilerplatey. Expressing "here's a line of source code, and the output should map 'blah' back to line X in the file" takes 13 lines of code, with an extra 6 per assertion.

It's always a balance to avoid obscuring the test too much, but maybe this would be a good place to write something like:

```ts
const TESTS = [
  {description: '...', source: '...', assertions: [{text: '...', line: ...}, ...],
]
```